pkg/controller/provider/controller.go

```diff
@@ -184,6 +200,24 @@ func (r Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (r
 		}
 	}
 
+	if provider.Type() == api.Ova {
+
+		deploymentName := fmt.Sprintf("%s-deployment-%s", ovaServerPrefix, provider.UID)
```

Comment:
```suggestion
		deploymentName := fmt.Sprintf("%s-deployment-%s", ovaServerPrefix, provider.Name)
```

Reply:
I've changed it to provider.Name so will be consistent with the rest of the naming.